Documents/TechnicalSpecifications/technicalSpecifications.md

```diff
@@ -162,4 +211,275 @@ Whitespace<br>
+      │     └─── Player 
+      │           └─── (player model with camera for top-down view)
+      └─── Sprites
+            ├─── CubeTextures
+            │     └─── (textures for the cube models)
+            ├─── Player
+            │     └─── (player sprites)
+            └─── Fonts
+                  └─── (fonts)
+```
+
+## Program structure
+
+### Map creation
+
+While our game is created using two different mode of camera, one being a top-down view and a side view, the maps will be made differently depending on the view that will be used for said map.<br>
+So, the maps will be created using 3D for the top-down view, and complete 2D for the side view.
+For the top-down view, the maps will be created using blocks and slopes. Buildings, environmental assets and props will be 2D sprites.
+For the side view, maps will be in complete 2D with multiple layers. One for the collisions with the player, one for the interactables and a last one for the background. 
+
+### User inputs
+
+For the player to move, interact and more, inputs have to be set. Here is the list of the different inputs for the game:
+
+![inputs.png](img/inputs.png)
+###### *Other inputs could be added later in the development phase*
+
+Those set buttons are done using a QWERTY keyboard<sup><a href="#3">[3]</a></sup>
+
+### Player movement
+
+Because of two different camera view, the movement available for the player will be different.<br>
+For the top-down view, the player will be able to move forward, backward, left and right. Here is the movement handling, where we select first the direction, then we apply the movement speed and the gravity:
+
+```GDscript
+var input_dir = Vector3.ZERO
+if Input.is_action_pressed("move_forward"):
+	input_dir.z -= 1
+if Input.is_action_pressed("move_backward"):
+	input_dir.z += 1
+if Input.is_action_pressed("move_left"):
+	input_dir.x -= 1
+if Input.is_action_pressed("move_right"):
+	input_dir.x += 1
+
+input_dir = input_dir.normalized() # Not going faster diagonally
+velocity.x = input_dir.x * MOVE_SPEED
+velocity.z = input_dir.z * MOVE_SPEED
+velocity.y = -10 # How fast to snap on floor
+```
+
+The player will have to use the 'W', 'A', 'S' and 'D' buttons on a QWERTY keyboard.
+
+For the side view, the player will loose the forward and backward movements, but be able the jump. Here is the sideway movement handling:
+
+```GDscript
+var direction := Input.get_axis("move_left", "move_right")
+if direction:
+	velocity.x = direction * SPEED
+else:
+	velocity.x = move_toward(velocity.x, 0, SPEED)
+```
+
+And here is the jump handling:
+
+```GDscript
+# Handle jump.
+if Input.is_action_just_pressed("jump") and is_on_floor():
+	velocity.y = JUMP_VELOCITY
+
+# Add the gravity.
+if not is_on_floor():
+	velocity += get_gravity() * delta
+```
+
+The player will have to use the 'A', 'D' and 'Space' buttons on a QWERTY keyboard.
+
+For each type of movement, a last function named 'move_and_slide()' is called to apply the movement.
```

Comment:
precise that the function is built-in in Godot

Reply:
done 👍